_tests_/pet.test.js

```diff
@@ -0,0 +1,155 @@
+  it('returns an object', () => {
+    expect(new Pet()).toBeInstanceOf(Object);
+  });
+});
+
+describe('constructor', () => {
+  it('sets the name property', () => {
+    const pet = new Pet('fido');
+
+    expect(pet.name).toEqual('fido');
+    expect(pet.hunger).toEqual(0);
+    expect(pet.fitness).toEqual(10);
+  });
+});
+
+describe('growUp', () => {
+  it('has a initial age of 0', () => {
+    const pet = new Pet('fido');
+
+    expect(pet.age).toEqual(0);
+  });
+});
+
+describe('growUp and getting unhealthier', () => {
+  it('increases the age of our pet by 1', () => {
+    const pet = new Pet('fido');
+    pet.growUp();
+
+    expect(pet.age).toEqual(1);
+    expect(pet.hunger).toEqual(5);
+    expect(pet.fitness).toEqual(7);
+  });
+
+  it("growup does not function if pet is not alive" ,() => {
+    const pet = new Pet("fido");
+    pet.age = 30;
+    expect(() => pet.growUp()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('walk', () => {
+  it('increases fitness by 4 to a maximum of 10', () => {
+    const pet = new Pet('fido');
+
+    pet.fitness = 8;
+    pet.walk();
+
+    expect(pet.fitness).toEqual(10);
+  });
+  it("walk does not function if pet is not alive" ,() => {
+    const pet = new Pet("fido");
+    pet.fitness = -1;
+    expect(() => pet.walk()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('feed', () => {
+  it('decreases the pets hunger levels', () => {
+    const pet = new Pet('fido');
+
+    pet.hunger = 0;
+    pet.feed();
+
+    expect(pet.hunger).toEqual(0);
+  });
+  it("feed does not function if pet is not alive" ,() => {
+    const pet = new Pet("Fido");
+    pet.hunger = 30;
+    expect(() => pet.feed()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('checkUp', () => {
+  it('check if pet needs a walk', () => {
+    const pet = new Pet('fido');
+
+    pet.fitness = 3
+    expect(pet.checkUp()).toBe('I need a walk');
+  });
+  it('check if pet is hungry', () => {
+    const pet = new Pet('fido');
+    pet.hunger = 5;
+    expect(pet.checkUp()).toEqual('I am hungry');
+   });
+  it('check if pet needs both walking and feeding', () => {
+    const pet = new Pet('fido');
+    pet.fitness = 3;
+    pet.hunger = 6;
+    expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+   });
+   it('if pet is neither hungry nor in need of a walk', () => {
+    const pet = new Pet('fido');
+   pet.fitness = 6;
+   pet.hunger = 3;
+   expect(pet.checkUp()).toEqual('I feel great'); 
+  });
+  it("checkup does not function if pet is not alive" ,() => {
+    const pet = new Pet("Fido");
+    pet.age = 30;
+    expect(() => pet.growUp()).toThrow("your pet is no longer alive :(");
+  });
+});
+
+describe('death', () => {
+  it('should be alive when growing up and walking', () => {
+    const pet = new Pet('Fido');
+    pet.growUp();
+    expect(pet.isAlive).toEqual(true);
+
+    pet.walk();
+    expect(pet.isAlive).toEqual(true);
+  });
```

Comment:
I would rewrite these test. You already have tests to show that an error is thrown when calling `growUp` and `walk` if the pet is dead. Tests related to the `isAlive` getter should only test stuff directly related to it, i.e. the conditions that make it return `true` or `false`. I would write this test like:
```
  it('isAlive returns true if fitness is more than 0, hunger is less than 10 and age is less than 30', () => {
    const pet = new Pet('fido');
    pet.fitness = 1;
    pet.hunger = 9;
    pet.age = 29;

    expect(pet.isAlive).toEqual(true);
  });
```